model-api-gen-gradle/src/main/kotlin/org/modelix/metamodel/gradle/UseExportedLanguagesTask.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2023.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.metamodel.gradle
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+import org.modelix.metamodel.generator.LanguageFiltering
+import org.modelix.metamodel.generator.ReadonlyProcessedLanguageSet
+import org.modelix.metamodel.generator.UnstableGeneratorFeature
+import javax.inject.Inject
+
+private typealias ActionWithLanguages = (ReadonlyProcessedLanguageSet) -> Unit
+
+/**
+ * This task allows to execute custom build steps which work with the generated and processed languages
+ * from the "exportMetaModelFromMps" task.
+ *
+ * The languages can be accessed through the [ReadonlyProcessedLanguageSet] structure.
+ * Custom user actions have to be registered as [ActionWithLanguages]] in [doWithLanguages]
+ */
+@CacheableTask
+@UnstableGeneratorFeature
+abstract class UseExportedLanguagesTask @Inject constructor(of: ObjectFactory) : DefaultTask() {
```

Comment:
I am not 100 % happy with the name of the task. "Use" somehow doesn't communicate well the intention of this task. But the ideas I can come up with are also not too good:

* WithExportedLanguagesTask
* PerformWithExportedLanguagesTask

Reply:
I liked `PerformWithExportedLanguagesTask`.

It's an @UnstableGeneratorFeature, so we can always rename it 😆

Reply:
Still, if we can avoid renaming things, people are happier usually ;-)